lang/cs/Org.Apache.REEF.IMRU/OnREEF/Client/REEFIMRUClient.cs

```diff
@@ -0,0 +1,123 @@
+            _configurationSerializer = configurationSerializer;
+            _jobSubmissionBuilderFactory = jobSubmissionBuilderFactory;
+        }
+
+        /// <summary>
+        /// Submits the job to reefClient
+        /// </summary>
+        /// <param name="jobDefinition">IMRU job definition given by the user</param>
+        /// <returns>Null as results will be later written to some directory</returns>
+        public IEnumerable<TResult> Submit(IMRUJobDefinition jobDefinition)
+        {
+            const string driverId = "BroadcastReduceDriver";
+
+            // The driver configuration contains all the needed bindings.
+            var imruDriverConfiguration = TangFactory.GetTang().NewConfigurationBuilder(new[]
+            {
+                DriverConfiguration.ConfigurationModule
+                    .Set(DriverConfiguration.OnEvaluatorAllocated,
+                        GenericType<IMRUDriver<TMapInput, TMapOutput, TResult>>.Class)
+                    .Set(DriverConfiguration.OnDriverStarted,
+                        GenericType<IMRUDriver<TMapInput, TMapOutput, TResult>>.Class)
+                    .Set(DriverConfiguration.OnContextActive,
+                        GenericType<IMRUDriver<TMapInput, TMapOutput, TResult>>.Class)
+                    .Set(DriverConfiguration.OnTaskCompleted,
+                        GenericType<IMRUDriver<TMapInput, TMapOutput, TResult>>.Class)
+                    .Build(),
+                jobDefinition.PartitionedDatasetConfgiuration
+            })
+                .BindNamedParameter(typeof(SerializedMapConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapFunctionConfiguration))
+                .BindNamedParameter(typeof(SerializedUpdateConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.UpdateFunctionConfiguration))
+                .BindNamedParameter(typeof(SerializedMapInputCodecConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapInputCodecConfiguration))
+                .BindNamedParameter(typeof(SerializedMapInputPipelineDataConverterConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapInputPipelineDataConverterConfiguration))
+                .BindNamedParameter(typeof(SerializedUpdateFunctionCodecsConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.UpdateFunctionCodecsConfiguration))
+                .BindNamedParameter(typeof(SerializedMapOutputPipelineDataConverterConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapOutputPipelineDataConverterConfiguration))
+                .BindNamedParameter(typeof(SerializedReduceConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.ReduceFunctionConfiguration))
+                .Build();
+
+            IConfiguration groupCommDriverConfig = TangFactory.GetTang().NewConfigurationBuilder()
```

Comment:
If you create this earlier, it can be part of the giant merge call above and doesn't need its own variable.